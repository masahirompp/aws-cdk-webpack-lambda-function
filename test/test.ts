import { Template } from "aws-cdk-lib/assertions";
import { App } from "aws-cdk-lib";
import { TestStack } from "./TestStack";

// CDK generate asset with a generated ID that we want to ignore from the comparison
expect.addSnapshotSerializer({
  test: (val) => typeof val === "string",
  print: (val) => {
    const newVal = (val as string).replace(
      /([A-Fa-f0-9]{64})(\.zip)/,
      "[HASH REMOVED]"
    );
    return `"${newVal}"`;
  },
});

test("CloudFormation Test", () => {
  // prepare
  const stack = new TestStack(new App(), "TestStack");
  const resources = Template.fromStack(stack).toJSON()["Resources"];
  expect(resources).toMatchSnapshot();
});
