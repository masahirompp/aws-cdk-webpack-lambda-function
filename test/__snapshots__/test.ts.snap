// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudFormation Test 1`] = `
Object {
  "SingletonLambdabe82c13fa959483791d71a3aabb2626aEEBB681F": Object {
    "DependsOn": Array [
      "SingletonLambdabe82c13fa959483791d71a3aabb2626aServiceRole406DEDB3",
    ],
    "Properties": Object {
      "Code": Object {
        "S3Bucket": Object {
          "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
        },
        "S3Key": "[HASH REMOVED]",
      },
      "Handler": "testFunction.handler",
      "Role": Object {
        "Fn::GetAtt": Array [
          "SingletonLambdabe82c13fa959483791d71a3aabb2626aServiceRole406DEDB3",
          "Arn",
        ],
      },
      "Runtime": "nodejs14.x",
    },
    "Type": "AWS::Lambda::Function",
  },
  "SingletonLambdabe82c13fa959483791d71a3aabb2626aServiceRole406DEDB3": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "Service": "lambda.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "ManagedPolicyArns": Array [
        Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
            ],
          ],
        },
      ],
    },
    "Type": "AWS::IAM::Role",
  },
  "WebpackFunction81CF8DFB": Object {
    "DependsOn": Array [
      "WebpackFunctionServiceRole9A70057E",
    ],
    "Properties": Object {
      "Code": Object {
        "S3Bucket": Object {
          "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
        },
        "S3Key": "[HASH REMOVED]",
      },
      "Handler": "testFunction.handler",
      "Role": Object {
        "Fn::GetAtt": Array [
          "WebpackFunctionServiceRole9A70057E",
          "Arn",
        ],
      },
      "Runtime": "nodejs14.x",
    },
    "Type": "AWS::Lambda::Function",
  },
  "WebpackFunctionServiceRole9A70057E": Object {
    "Properties": Object {
      "AssumeRolePolicyDocument": Object {
        "Statement": Array [
          Object {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": Object {
              "Service": "lambda.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "ManagedPolicyArns": Array [
        Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
            ],
          ],
        },
      ],
    },
    "Type": "AWS::IAM::Role",
  },
}
`;
